events/git-and-github.md

```diff
@@ -0,0 +1,129 @@
+
+4. **A demo**
+
+   A small demo of how a git repository is initialised. Creating files and adding them to staging area. Commit the changes. While you are typing the commands, do explain the audience what these commands do. For example, while typing `git add FILE` , tell them that now this will add `FILE` to the staging area.
+
+   Though any demo works, it is a good idea to demonstrate something that's closer to daily life examples. For demo purpose, you can create a project as a "Facebook clone". You don't have to write any real code. You can simply create a text file and and something to demonstrate that some X feature has been added, and commit it. Audience will indirectly absorb the idea the commits are made after every feature added or after every bug is fixed.
+
+5. **Stop and Revise**
+
+   It is a good time to stop and revise the concepts. Staging area is a non-intuitive concept. Try using analogies to make the idea clear(For analogies [refer answer by Spoike](https://softwareengineering.stackexchange.com/questions/119782/what-does-stage-mean-in-git)). Ask the audience if they have any doubts. In an offline session TAs can individually clear people's queries. 
+
+6. **Reverting using Git**
+
+   This is better done using demo. Show them the following
+
+   - How to undo changes made after a commit using `git checkout`
+   - How to remove changes from staging area
+   - How to go back commits using `git reset --hard`
+
+   (This can be done using a story. For example, an evil cat makes changes in your projects and you use git commands to revert those changes)
+
+7. **Branching**
+
+   Branching is another non-intuitive concept. You can start with a problem of how difficult it is to have multiple copies of a project with multiple people working on it. The problem should be posed in such a way that its ultimate solution has to be Git Branching. This will help them better understand why branches are important. You can show the commit trees of different branches and how they diverge. Also show how the tree merges after git merge.
+
+   Tools that can be used - [https://git-school.github.io/visualizing-git/](https://git-school.github.io/visualizing-git/)
+
+8. **Branching demo**
+
+   You can use the "Facebook clone" project for the demo. Make a branch for a new feature. Make some commits in the new branch and show the commits. Jump to `master` branch and show that those commits are accept. Merge the newly created branch and show how commits have appeared. 
```

Comment:
*accepted

Reply:
Sorry, I meant to write "absent". Changed it! Thanks